agent/command/results_xunit_parser.go

```diff
@@ -106,6 +108,14 @@ func (tc testCase) toModelTestResultAndLog(conf *internal.TaskConfig) (testresul
 	// Replace spaces, dashes, etc. with underscores.
 	res.TestName = util.CleanForPath(res.TestName)
 
+	if math.IsNaN(float64(tc.Time)) {
+		logger.Task().Errorf("Test '%s' time was NaN, its calculated duration will be incorrect", res.TestName)
```

Comment:
Is this supposed to transform the NaN into a zero value as well?

Reply:
No because `NaN` already results in the `res.TestEndTime` below equating to whatever `TestStartTime` was, so we don't need to modify the time value here to prevent `TestEndTime` from evaluating to something strange